src/plugins/simulator/fsxcommon/simulatorfsxcommon.cpp

```diff
@@ -1998,7 +2003,13 @@ namespace BlackSimPlugin::FsxCommon
                 // update situation
                 if (forceUpdate || !this->isEqualLastSent(result.getInterpolatedSituation()))
                 {
-                    SIMCONNECT_DATA_INITPOSITION position = this->aircraftSituationToFsxPosition(result, sendGround);
+                    // adjust altitude to compensate for FS2020 temperature effect
+                    CAircraftSituation situation = result;
+                    const CLength relativeAltitude = situation.geodeticHeight() - getOwnAircraftPosition().geodeticHeight();
+                    const double altitudeDeltaWeight = qBound(3000.0, relativeAltitude.abs().value(CLengthUnit::ft()), 6000.0) / 3000 - 1;
```

Comment:
This weight should be 1 for the ``relativeAltitude`` between 0-3000ft (fully apply offset) and linearly fade out till 6000ft.
If I am not mistaken it seems that the weight is 1 for 6.000ft only and decreases to reach 0 at 3000 ft.

Reply:
Good catch!